libraries/FFat/src/FFat.cpp

```diff
@@ -40,16 +40,20 @@ const esp_partition_t *check_ffat_partition(const char* label)
 
 bool F_Fat::begin(bool formatOnFail, const char * basePath, uint8_t maxOpenFiles, const char * partitionLabel)
 {
-    if(_wl_handle){
+    if(_wl_handle != WL_INVALID_HANDLE){
         log_w("Already Mounted!");
         return true;
```

Comment:
Ok, this case is if the Object has already been opened, Is there any way to verify that the `basePath` points to the previous open?  Also, does it point to the same `partitionLabel`?

I can see where someone would use `begin()` to change the current mount point/partition.  

Would you actually force a dismount before processing the mount?  

What happens to a fileObject when you dismount the file system?

Chuck.

Reply:
Forcing an end before another begin certainly takes care of closing any opened files, and seems pretty intuitive.  Unmount is handled by the fatfs unmount.  I expect that this would do unpleasant things to dependent objects. Should clearing the mountpoint signal FS(impl) to close all files?

Reply:
I can see a couple of paths:
* Flush/close all open files.  Cause every access to orphaned fileObject to return an error.  Make it painful for someone to make this mistake.
* Fail the `begin()` if the mount point is already created.

I can see people going to light sleep first dismounting (to reduce current consumption..) then re-mounting after wakeup.  How does deep Sleep affect it?  It seems to me the Deep Sleep is just a program controlled power off with a power on reset on wakeup?  If so, then Deep Sleep is equivalent to uncontrolled shutdown with a potential dirty volume on wakeup?

How much effort are you willing to make it "foolproof"?  I think you should use the second option.  Make a call to `begin()` with an active mount a fatal error.  Unmount(protect filesystem) then return fail.

Reply:
one of you please ping me after you have agreed :)

Reply:
btw @lbernstone, it seems this fix somehow breaks FFat.format(). I downloaded the revised FFat library files from your fork to replace the files in my local repo. The fix indeed solves #3244. But it seems to render FFat.format useless. It returns no error, but it doesn't erase the FFAT partition. Please check. You can use the library example.